### ISSUE TITLE
ActiveField form group has-error class missing when model attribute name is tabular (prefix [])

### ISSUE THREAD

**Issue by FabrizioCaldarelli** [yiisoft/yii2#15801]
### What steps will reproduce the problem?
Use tabular as attribute name for field:
```php
<?= $form->field($model, '[0]attributeOfModel')->textInput() ?>
```

Set form enableClientValidation to false.

### What is the expected result?
If there is an error on attributeOfModel, this should add an has-error class to container div (with 'form-group' class)

### What do you get instead?
The 'has-error' class is not added.

### Additional info

The problem is in `addErrorClassIfNeeded` of `yii\widget\ActiveField` class, because it checks hasErrors method on attribute and not on cleared attribute name.

This will fix:

```php
    protected function addErrorClassIfNeeded(&$options)
    {
        // This get correctly attribute name when attribute name is tabular. 
        $attributeName = yii\helpers\BaseHtml::getAttributeName($this->attribute);

        if ($this->model->hasErrors($attributeName)) {
            Html::addCssClass($options, $this->form->errorCssClass);
        }
    }
```

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.14.1
| PHP version      | 7.1.14
| Operating system | Centos 7

**Comment by SilverFire** [comment]
Thank you for the report. Would you like to create a Pull Request?

**Comment by FabrizioCaldarelli** [comment]
I've published the pull request: https://github.com/yiisoft/yii2/pull/15805

**Comment by SilverFire** [comment]
Fixed in #15805
